lib/src/wso2/utils-cdk.ts

```diff
@@ -31,9 +33,39 @@ export const addLambdaAndProviderForWso2Operations = (args: {
     wso2LambdaEntry = `${args.baseDir}/handler/index.js`;
   }
 
+  // vpc is undefined if no network is defined
+  let vpc: IVpc | undefined;
+  const { customResourceConfig } = args.props;
+
+  const securityGroups = [];
+
+  // Create security group for custom resource if VPC is defined and no security group is defined
+  if (args.props.customResourceConfig?.network) {
+    vpc = vpcFromConfig(args.scope, args.props.customResourceConfig.network);
+
+    // never use network configuration, only explicit VPC from previous step
+    // eslint-disable-next-line fp/no-delete
+    delete customResourceConfig?.network;
```

Comment:
suggestion: mutating a function parameter is bad practice, it can lead to confusion and make it harder to the caller to understand it.

As we never want to provide the network we could omit it from the `customResourceConfig`

```ts
const { network, ...customResourceConfig } = args.props.customResourceConfig ?? {};
```

or removing it from `BaseNodeJsFunction` props
```ts
const customResourceFunction = new BaseNodeJsFunction(args.scope, `${args.id}-custom-lambda`, {
    ...customResourceConfig,
    network: undefined,
```

Reply:
I agree! I was trying other alternatives but it was adding too much code. Thanks for the great suggestion!